### PR TITLE
Make std crate feature default for all ink crates

### PR DESCRIPTION
**Pull request by Robbepop**
Normalizes the `std` crate feature usage of ink! to be enabled by default.
To use `no_std` environment users can opt-out with `default-features = false` or `--no-default-features`.
Some dependencies have been updated.
The `std` feature flag is now correctly propagated to crate dependencies.

### CODE REVIEW

File: core/Cargo.toml [hunk 1]
```diff
@@ -18,14 +18,23 @@ categories = ["no-std", "embedded"]
...
 
 [features]
-default = []
+default = ["std"]
 test-env = ["std"]
 std = [
+	"ink_abi/std",
     "ink_alloc/std",
     "ink_utils/std",
+	"parity-codec/std",
```
Comment (ascjones) [inline core/Cargo.toml]: Indentation

File: core/Cargo.toml [hunk 2]
```diff
@@ -18,14 +18,23 @@ categories = ["no-std", "embedded"]
 include = ["/Cargo.toml", "src/**/*.rs", "/README.md", "/LICENSE"]
 
 [dependencies]
-ink_alloc = { path = "../alloc/" }
-ink_utils = { path = "../utils/" }
-parity-codec = { version = "3.2", default-features = false, features = ["derive", "full"] }
+ink_abi = { path = "../abi", default-features = false }
```
Comment (ascjones) [inline core/Cargo.toml]: abi not in master yet

File: core/Cargo.toml [hunk 3]
```diff
@@ -18,14 +18,23 @@ categories = ["no-std", "embedded"]
...
+ink_alloc = { path = "../alloc/", default-features = false }
+ink_utils = { path = "../utils/", default-features = false }
+parity-codec = { version = "4.1", default-features = false, features = ["derive", "full"] }
+
+[dependencies.type-metadata]
+git = "https://github.com/type-metadata/type-metadata.git"
+default-features = false
+features = ["derive"]
```
Comment (ascjones) [inline core/Cargo.toml]: this is for your abi PR?

File: lang/Cargo.toml [hunk 4]
```diff
@@ -39,10 +39,14 @@ name = "ink_lang"
 proc-macro = true
 
 [features]
-default = []
+default = ["std"]
 std = [
     "ink_utils/std",
+	"ink_model/std",
```
Comment (ascjones) [inline lang/Cargo.toml]: indentation

File: core/Cargo.toml (hunk 2 above)
Comment (Robbepop) [reply core/Cargo.toml]: thank god you found it, I was already suspicious about the CI error ...

File: core/Cargo.toml (hunk 3 above)
Comment (Robbepop) [reply core/Cargo.toml]: damn, I somehow accidentally based the initial PR on ink-abi branch and git was very unforgiving about that. thanks

**Comment by Robbepop** [comment]
The CI failure: `thread 'main' panicked at 'internal error: entered unreachable code', src/tools/rustfmt/src/formatting.rs:320:18`

**Pull request merged**